request/auth.php

```diff
@@ -14,7 +14,7 @@
                     case 'login':
                         if (!empty($_POST['email']) && !empty($_POST['password'])) {
                             if (FirstAide\Utils::isValidEmail($_POST['email'])) {
-                                $Auth = FirstAide\Authentication::withEmailPassword(
+                                $Auth = FirstAide\Authentication::createInstanceWithEmailPassword(
```

Comment:
@MehaKaushik 
``` isset($_POST['csrf_token']) && $_POST['csrf_token'] == $csrf_token```
If we move this section to a method, we will need to move POST and GET variables also inside the function, which is a bad practice. I totally agree that the code should be made clean, but not at the expense of violation of standards. These standards help us to ensure global variables and CRUD request variables are separated from the functions and methods(functions of a class)
This ensures functions and methods are treated as black boxes. In case we move them, functions will expect POST or GET method calls, making CLI access not possible and it will be a block in PHPUnit testing as well. 
Our primary goal is to make the code as much modular as possible according to standards. This does not even make the code reusable, as  in case of GET calls we will have to create another function.
I have changed this if statement in such a way that it can a validate a token. The CSRF token from POST is sent to the method as a parameter which is following all the standards, making the code, cleaner, modular and reusable. I hope I am able to make you understand your doubt.